_posts/2023-03-09-group-equivariant-cnn-1/group-equivariant-cnn-1.Rmd

```diff
@@ -0,0 +1,148 @@
+
+First, consider the domain. A $7$ is a $7$ wherever it appears on the grid. We thus need an operation that is *translation-equivariant*: It flexibly adapts to shifts (translations) in its input. More concretely, in our context, *equivariant* operations are able to detect some object's properties even if that object has been moved, vertically and/or horizontally, to another location. *Convolution*, ubiquitous not just in deep learning, is just such a shift-equivariant operation.
+
+Let me call special attention to the fact that, in equivariance, the essential thing is that "flexible adaptation" (as I casually termed it before). Translation-equivariant operations *do* care about an object's new position; they record a feature not abstractly, but at the object's new position. To see why this is important, consider the network as a whole. When we compose convolutions, we build a hierarchy of feature detectors. That hierarchy should be functional no matter where in the image; in addition, it has to be consistent: Location information needs to be preserved between layers.
+
+Terminology-wise, thus, it is important to distinguish equivariance from *invariance*. An invariant operation, in our context, would still be able to spot a feature wherever it occurs; however, it would happily forget where that feature happened to be. Clearly, then, to build up a hierarchy of features, translation-*invariance* is not enough.
+
+What we've done right now is derive a requirement from the domain, the input grid. What about the task? If, finally, all we're supposed to do is name the digit, now suddenly location does not matter anymore. In other words, once the hierarchy exists, invariance *is* enough. In neural networks, *pooling* is an operation that forgets about (spatial) detail: It only cares about the mean, say, or the maximum value itself. This is what makes it suited to "summing up" information about a region, or a complete image, if at the end we only care about returning a class label.
+
+In a nutshell, we were able to specify design desiderates based on (1) what we're given and (2) what we're tasked with.
+
+After this high-level sketch of Geometric Deep Learning, we zoom in on this series' of posts designated topic: *group-equivariant* convolutional neural networks.
+
+The why of "equivariant" should not, by now, pose too much of a riddle. What about that "group" prefix, though?
+
+## The "group" in group-equivariance
+
+As you may have guessed from the introduction, talking of "principled" and "math-driven", this *really* is about groups "in the math sense". Dependent on your background, the last time you heard about groups was in school, and with not even a hint at why they matter. I'm certainly not qualified to summarize the whole richness of what they're good for, but I hope that by the end of this post, their importance in deep learning will make intuitive sense.
```

Comment:
```suggestion
As you may have guessed from the introduction, talking of "principled" and "math-driven", this *really* is about groups in the "math sense." Depending on your background, the last time you heard about groups was in school, and with not even a hint at why they matter. I'm certainly not qualified to summarize the whole richness of what they're good for, but I hope that by the end of this post, their importance in deep learning will make intuitive sense.
```

Reply:
same error as years ago ... and I was sure I'd had it the right way round :-(